patch-image.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/bash
+set -ex
+
+# PATCH argument is in the form "project refs" where refs is a list of
+# comma-separated gerrit refs, fo example refs/changes/67/759567/1
+PROJ_NAME=$(echo $1 | cut -d " " -f1)
+REFSPECS=$(echo $1 | cut -d " " -f2)
+
+cd /tmp
+git clone "https://opendev.org/openstack/$PROJ_NAME"
+cd "$PROJ_NAME"
+for REFSPEC in $REFSPECS; do
+    git fetch "https://review.opendev.org/openstack/$PROJ_NAME" "$REFSPEC"
+    git -c user.name=Nobody -c user.email=nobody@example.com cherry-pick FETCH_HEAD
```

Comment:
This is cherry-picking patches on top of master, but it's not obvious that master is always what you want to be based on. (e.g. Zuul does optimistic merging, so if this were used in a Zuul gate test it wouldn't necessarily test the right thing.) It would seem simpler to just pass a single refspec and check that out. If you have multiple changes in a series, you can just point to the last one.

Reply:
For a Zuul CI job we'll need to pass a link to a local checkout, I wonder if we could make it possible.

Reply:
@zaneb so just using checkout instead of cherry-pick? that could work better for series, I agree. I tested this only on single patches and honestly didn't think about that.
@dtantsur we just need to pass the last refspec in the series, why a link ?